### ISSUE TITLE
`F821` forward reference false positive in stubs (`.pyi` files)

### ISSUE THREAD

**Issue by Avasam** [charliermarsh/ruff#3909]
Given this code:
```py
class A(B): ...
class B: ...
```

Does raise `F821 Undefined name` in python files, but should not raise in stubs `.pyi` files.

Whilst it is true that it's a best practice to avoid forward references, sometimes it can't be avoided in more complex cases. "Undefined name" is also simply not true in this context.

**Comment by charliermarsh** [comment]
Possibly the same as #3011.

**Comment by charliermarsh** [comment]
(I think we need to implement different visitation rules for `.pyi` files.)

**Comment by Avasam** [comment]
> Possibly the same as #3011.

It does look like I duplicated that issue. Sorry I do try to search first, I must've mistyped the rule code as no results came up. Feel free to close this in favor of #3011

**Comment by charliermarsh** [comment]
No worries, sadly I have encyclopedic knowledge of our issues so it's easy for me to recall duplicates :joy:
